python/lsst/sims/utils/coordinateTransformations.py

```diff
@@ -103,7 +62,194 @@ def calcLmstLast(mjd, longRad):
+
+    _de2hOutput=palpy.de2h(hourAngle, dec,  latitude)
+
+    #return (altitude, azimuth)
+    return _de2hOutput[1], _de2hOutput[0]
+
+
+def galacticFromEquatorial(ra, dec):
+    '''Convert RA,Dec (J2000) to Galactic Coordinates
+
+    All angles are in radians
+
+    @param [in] ra is right ascension in radians, either a float or a numpy array
+
+    @param [in] dec is declination in radians, either a float or a numpy array
+
+    @param [out] gLong is galactic longitude in radians
+
+    @param [out] gLat is galactic latitude in radians
+    '''
+
+    if isinstance(ra, numpy.ndarray):
+        gLong, gLat = palpy.eqgalVector(ra, dec)
+    else:
+        gLong, gLat = palpy.eqgal(ra, dec)
+
+    return gLong, gLat
+
+
+def equatorialFromGalactic(gLong, gLat):
+    '''Convert Galactic Coordinates to RA, dec (J2000)
+
+    @param [in] gLong is galactic longitude in radians, either a float or a numpy array
+
+    @param [in] gLat is galactic latitude in radians, either a float or a numpy array
+
+    @param [out] ra is right ascension in radians
+
+    @param [out] dec is declination in radians
+    '''
+
+    if isinstance(gLong, numpy.ndarray):
+        ra, dec = palpy.galeqVector(gLong, gLat)
+    else:
+        ra, dec = palpy.galeq(gLong, gLat)
+
+    return ra, dec
+
+
+def cartesianFromSpherical(longitude, latitude):
+    """
+    Transforms between spherical and Cartesian coordinates.
+
+    @param [in] longitude is the input longitudinal coordinate
+
+    @param [in] latitude is the input latitudinal coordinate
+
+    @param [out] a list of the (three-dimensional) cartesian coordinates on a unit sphere
+
+    All angles are in radians
+    """
+
+    cosDec = numpy.cos(latitude)
+    return numpy.array([numpy.cos(longitude)*cosDec,
+                      numpy.sin(longitude)*cosDec,
+                      numpy.sin(latitude)])
+
+
+def sphericalFromCartesian(xyz):
+    """
+    Transforms between Cartesian and spherical coordinates
+
+    @param [in] xyz is a list of the three-dimensional Cartesian coordinates
+
+    @param [out] returns longitude and latitude
+
+    All angles are in radians
+    """
+
+    rad = numpy.sqrt(xyz[:][0]*xyz[:][0] + xyz[:][1]*xyz[:][1] + xyz[:][2]*xyz[:][2])
+
+    longitude = numpy.arctan2( xyz[:][1], xyz[:][0])
+    latitude = numpy.arcsin( xyz[:][2] / rad)
+
+    return longitude, latitude
+
+
+def rotationMatrixFromVectors(v1, v2):
+    '''
+    Given two vectors v1,v2 calculate the rotation matrix for v1->v2 using the axis-angle approach
+
+    @param [in] v1, v2 are two Cartesian vectors (in three dimensions)
+
+    @param [out] rot is the rotation matrix that rotates from one to the other
+
+    '''
+
+    # Calculate the axis of rotation by the cross product of v1 and v2
+    cross = numpy.cross(v1,v2)
+    cross = cross / numpy.sqrt(numpy.dot(cross,cross))
+
+    # calculate the angle of rotation via dot product
+    angle  = numpy.arccos(numpy.dot(v1,v2))
```

Comment:
this method does not make sense if you are not calling it on unit vectors (it is trying to find a rotation from one vector to the other).  I added code so that it raises an exception if you do not pass it unit vectors

(I guess you could pass it non-unit vectors, as long as they are of the same magnitude... I am going to ignore that use case)